apps/dh/api-dh/source/DataHub.WebApi.Tests/Modules/Charges/ChargeStatusTests.cs

```diff
@@ -0,0 +1,118 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Energinet.DataHub.WebApi.Modules.Charges.Models;
+using Energinet.DataHub.WebApi.Tests.Extensions;
+using Energinet.DataHub.WebApi.Tests.Mocks;
+using Energinet.DataHub.WebApi.Tests.TestServices;
+using HotChocolate.Execution;
+using Moq;
+using Xunit;
+
+namespace Energinet.DataHub.WebApi.Tests.Modules.Charges;
+
+public class ChargeStatusTests
+{
+    private static readonly string _query =
+    $$"""
+    {
+        chargesByPeriod(
+            query: { from: "2020-01-01T00:00:00Z", to: "2020-12-31T23:59:59Z" }
+            after: null
+            before: null
+            first: 10
+            last: null
+            order: {
+                id: ASC
+            }
+        ) {
+            totalCount
+            nodes {
+                status
+            }
+        }
+    }
+    """;
+
+    private static readonly DateTimeOffset _sameDate = DateTimeOffset.Now.AddDays(5);
+
+    public static IEnumerable<object[]> GetTestCases()
+    {
+        yield return new object[] { "Cancelled same date", _sameDate, _sameDate, false };
+        yield return new object[] { "Closed", DateTimeOffset.Now.AddDays(-4), DateTimeOffset.Now.AddDays(-4), true };
+        yield return new object[] { "Current", DateTimeOffset.Now.AddDays(-5), DateTimeOffset.Now.AddDays(10), true };
+        yield return new object[] { "Missing prices series with end", DateTimeOffset.Now.AddDays(-5), DateTimeOffset.Now.AddDays(2), false };
+    }
+
+    [Theory]
+    [MemberData(nameof(GetTestCases))]
+    public async Task ChargesStatus(string testname, DateTimeOffset validFrom, DateTimeOffset validTo, bool hasAnyPrices) =>
+        await ExecuteTestAsync(testname, validFrom, validTo, hasAnyPrices);
+
+    [Fact]
+    public async Task ChargeStatus_MissingPricesSeriesWithoutEndDate()
+    {
+        await ExecuteTestAsync("Missing prices series end", DateTimeOffset.Now.AddDays(-5), null, false);
+    }
+
+    private static async Task ExecuteTestAsync(string testname, DateTimeOffset validFrom, DateTimeOffset? validTo, bool hasAnyPrices)
+    {
+        var server = new GraphQLTestService();
+
+        server.ChargesClientMock
+            .Setup(x => x.GetChargesByPeriod(It.IsAny<GetChargesByPeriodQuery>()))
+            .Returns(
+            [
+                new ChargeDto(
+                    Id: Guid.NewGuid(),
```

Comment:
question: will this not cause the snapshots to change between tests? I usually use a static GUID for cases like this.

Reply:
I dont include the id in the query result